src/System.Text.Json/tests/JsonEncodedTextTests.cs

```diff
@@ -280,17 +280,20 @@ public static void InvalidEncode()
         [OuterLoop]
         public static void InvalidLargeEncode()
         {
-            char[] largeValue = new char[400_000_000];
-
-            Span<char> largeValueSpan = largeValue.AsSpan();
-            largeValueSpan.Fill('a');
-
-            var largeValueString = new string(largeValue);
-            byte[] utf8Value = Encoding.UTF8.GetBytes(largeValue);
+            try
+            {
+                var largeValueString = new string('a', 400_000_000);
+                var utf8Value = new byte[400_000_000];
+                utf8Value.AsSpan().Fill((byte)'a');
 
-            Assert.Throws<ArgumentException>(() => JsonEncodedText.Encode(largeValueString));
-            Assert.Throws<ArgumentException>(() => JsonEncodedText.Encode(largeValue));
-            Assert.Throws<ArgumentException>(() => JsonEncodedText.Encode(utf8Value));
+                Assert.Throws<ArgumentException>(() => JsonEncodedText.Encode(largeValueString));
+                Assert.Throws<ArgumentException>(() => JsonEncodedText.Encode(largeValueString.AsSpan()));
+                Assert.Throws<ArgumentException>(() => JsonEncodedText.Encode(utf8Value));
+            }
+            catch (OutOfMemoryException)
```

Comment:
Consider instead using a conditional theory that checks if the process is 64-bit: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs#L25